### ISSUE TITLE
Test Builds #9

### ISSUE THREAD

**Issue by Mikahlo** [Spookcity/ROMS-G800F#25]
A continuation of [#10](https://github.com/Spookcity/ROMS-G800F/issues/10), [#18](https://github.com/Spookcity/ROMS-G800F/issues/18), [#19](https://github.com/Spookcity/ROMS-G800F/issues/19), [#20](https://github.com/Spookcity/ROMS-G800F/issues/20), [#21](https://github.com/Spookcity/ROMS-G800F/issues/21), [#22](https://github.com/Spookcity/ROMS-G800F/issues/22), [#23](https://github.com/Spookcity/ROMS-G800F/issues/23) and [#24](https://github.com/Spookcity/ROMS-G800F/issues/24)

**Comment by Oleggluk** [comment]
From telegrams all were expelled and here there is silence.

**Comment by friedensfurz** [comment]
New test build of RR online here (https://androidfilehost.com/?fid=962339331459002251)@Oleggluk . Please test and let me know if there are any problems ;)

**Comment by Oleggluk** [comment]
@friedensfurz Looks pretty stable. I have time 1.40 nights, I can not test for a long time. I moved from RR5.8.5 to RR6.1 without formatting Date section, all applications are saved, except contacts !!! (before the firmware I deleted only Magisk and xposed). I also did a wipe system, cache and dalwick cache. In RR6.1 there are no physical button settings, but my button settings have been saved from RR5.8.5!. The camera does not work at all. Snap camera only takes pictures. The Internet works, wi-fi works, vpn works .. I really want to sleep))

**Comment by friedensfurz** [comment]
@Oleggluk Thanks for testing that. Is the camera really not working at all? And the main thing, are Apps working fine ? Some people have a new storage issue on Lineage and it is not possible to modify any file, only creating and deleting works.  If there is an issue, the call history will be emptied after every call. Thanks for testing and if there's any other issues, just write them here ;)

**Comment by Oleggluk** [comment]
@friedensfurz Concerning the problems of changing files, I noticed only one problem, in X-plore I can not edit text files, but this is possible with other file managers. The camera in my cell photographs gray squares. Snap camera takes pictures well. Video shooting does not work anywhere. I can not check the history of calls, I'm an Internet maniac)), calls are not for me)).

**Comment by Oleggluk** [comment]
@friedensfurz There is no problem with the history of the call. But I do not see any advantages from using android 8.1. For me, android 7.1.2 is more suitable and more familiar.
I now want to install ResurrectionRemix-M-v5.7.4-BlackMix-20160921-kminilte.zip)) for the sake of experiment. And then I'll return to RR 5.8.5 (android 7.1.2).

**Comment by friedensfurz** [comment]
@Oleggluk Thanks for testing that out. One last thing if you (or someone else) tests RR-O again, are heavy camera apps like Snapchat etc. working? I'd be interested if they crash the camera or not. Thanks ;)

**Comment by Oleggluk** [comment]
@friedensfurz The company Samsung announced a new budget smartphone on Android 8.1 Oreo from the Android Go project. The characteristics are similar to our device, only we have the Exynos 3 Quad 3470, and the new Exynos 7570 with a frequency of 1.4 GHz. Maybe it can be ported to our phone? Or use some files to complement the Los15.1 and RR-v6.1?
![gpiyomf4yyukxz3rgupj9irr6nqwan](https://user-images.githubusercontent.com/27913216/41854304-8d3e1ede-7898-11e8-836a-e00ee183515d.jpg)

**Comment by Spookcity** [comment]
Hey guys! (Assuming anyone still visits these parts) It's been awhile,but I'm still around and am still using my S5 Mini. Just a small update along with some ROM builds I've done in the last week or so that I figured I would share.
I've not had time for much Android wise recently,but I did spend a few days trying to solve the video recording problem on Oreo ROMS. I had no luck sadly. I must assume it is a device specific problem as I believe video recording was sorted out some time ago,so thats where I focused,but still no luck.

**ROMS**
AospExtended-v5.8 - [AEX 5.8_microG](https://androidfilehost.com/?fid=1322778262904009865)

**READ ME**
I have fully integrated [MicroG](https://microg.org) into this ROM. DO NOT flash a Gapps package! If you do not know what this is or do not wish to use it I suggest another ROM. I have included the [Core](https://github.com/microg/android_packages_apps_GmsCore/wiki) obviously,but along with that I have also built in the other parts

- [MapsAPI](https://github.com/microg/android_frameworks_mapsv1)
- [FakeStore](https://github.com/microg/android_packages_apps_FakeStore)
- NominatimNlpBackend (Address lookup backend)
- [IchnaeaNlpBackend](https://github.com/microg/IchnaeaNlpBackend)
**These need to be setup with MicroG settings app**
**You also must go to settings -> Apps and Notifications -> App Permissions -> Spoof Package Signature. Click the 3 dot menu -> Show system and the check both FakeStore and MicroG**
- If you don't use F-droid and/or APK Pure or similar for your app needs and would like to install the play store you can have a look [HERE](https://shadow53.com/android/downloads/microg/)

**Other ROMS**
XenonHD - [XenonHD-180910-Unofficial-kminilte](https://androidfilehost.com/?fid=1322778262904009867)
AICP -[aicp_kminilte_o-13.1-UNOFFICIAL-20180908](https://androidfilehost.com/?fid=1322778262904009864)
GZOSP - [Gzosp-kminilte-8.1.0.Stable.1.1-UNOFFICIAL-20180907-1121](https://androidfilehost.com/?fid=1322778262904009866)

I merged the latest Lineage 15.1 sources into my Hardware/Samsung folder for all of these builds as it was needed to build a few of these. Did it break things? I don't know. I have only tested AEX,but please let me know if so.

Maybe I will have more time in the coming months to mess with Android more. Here's hoping!

Enjoy!

**Comment by Mikahlo** [comment]
Hey @Spookcity : which filesystem do you use for the internal storage? I can only copy stuff to the external sd-card but not to the internal storage. I use f2fs as internal filesystem and vfat for external.

**Comment by Mikahlo** [comment]
Can't copy stuff with the file manager or with MiX. Already tried different file systems. Guess I'm SOL xD. The AEX boot animation looks nice btw.
`09-21 20:10:54.686 I/MiX     ( 4582): ----- Operation started ----------------------
09-21 20:10:54.688 I/MiX     ( 4582): Action: COPY
09-21 20:10:54.689 I/MiX     ( 4582): Calculating...
09-21 20:10:54.691 I/MiX     ( 4582): Total: 4268787  bytes
09-21 20:10:54.699 I/OperationHelper( 4582): Copy from /storage/emulated/0/Download/Magisk-v17.1(17100).zip
09-21 20:10:54.699 I/OperationHelper( 4582): To /storage/emulated/0/Pictures
09-21 20:10:54.708 D/OH:     ( 4582): default
09-21 20:10:54.731 D/SurfaceFlinger( 2097): duplicate layer name: changing com.mixplorer/com.mixplorer.activities.BrowseActivity to com.mixplorer/com.mixplorer.activities.BrowseActivity#1
09-21 20:10:54.731 D/MiX     ( 4582): Os Why?!
09-21 20:10:54.743 E/OperationHelper( 4582): java.io.FileNotFoundException
09-21 20:10:54.744 W/GrallocMapperPassthrough( 2097): buffer descriptor with invalid usage bits 0x400
09-21 20:10:54.750 I/MiX     ( 4582): Operation failed: No item copied
09-21 20:10:54.754 I/MiX     ( 4582): ----- Operation failed. 0s -------------
09-21 20:10:54.762 W/GrallocMapperPassthrough( 2097): buffer descriptor with invalid usage bits 0x400
09-21 20:10:54.795 W/GrallocMapperPassthrough( 2097): buffer descriptor with invalid usage bits 0x400
09-21 20:10:54.836 E/memtrack( 2253): Couldn't load memtrack module
09-21 20:10:54.836 W/android.os.Debug( 2253): failed to get memory consumption info: -1
09-21 20:10:54.891 W/GrallocMapperPassthrough( 2097): buffer descriptor with invalid usage bits 0x400
09-21 20:10:54.903 W/GrallocMapperPassthrough( 2097): buffer descriptor with invalid usage bits 0x400
09-21 20:10:55.002 W/GrallocMapperPassthrough( 2097): buffer descriptor with invalid usage bits 0x400
09-21 20:10:56.780 I/ActivityManager( 2253): START u0 {flg=0x10804000 cmp=com.android.systemui/.recents.RecentsActivity} from uid 10025
09-21 20:10:56.793 W/GrallocMapperPassthrough( 2456): buffer descriptor with invalid usage bits 0x400
09-21 20:10:56.793 I/MiX     ( 4582): MiX paused.
09-21 20:10:56.794 E/BufferQueueProducer( 2456): [unnamed-2456-8] setMaxDequeuedBufferCount: 3 dequeued buffers would exceed the maxBufferCount (2) (maxAcquired 1 async 0 mDequeuedBufferCannotBlock 0)
09-21 20:10:56.794 E/Surface ( 2456): IGraphicBufferProducer::setBufferCount(4) returned Invalid argument
09-21 20:10:56.811 E/BufferItemConsumer( 2456): [unnamed-2456-8] Failed to release buffer: Unknown error -1 (1)
09-21 20:10:56.866 W/GrallocMapperPassthrough( 2097): buffer descriptor with invalid usage bits 0x400
09-21 20:10:56.870 W/GrallocMapperPassthrough( 2456): buffer descriptor with invalid usage bits 0x400`

**Comment by aelfinn** [comment]
Some quick feedback on the AICP, GZOSP, and XenonHD builds: AICP runs pretty well, but hardware button backlight seems to be non-functional; I could get GZOSP to boot only without GApps or Magisk; XenonHD seems to be running really well, with only LiveDisplay apparently not working (selection is not retained) and very little standby battery drain with Magisk module “Enable Doze for GMS”.

**Comment by Spookcity** [comment]
@Mikahlo Yes,I have only just recently noticed this strange file system issue. I was using AEX nougat until just a week or so ago. I am not real sure where the problem lies,but it seems as if it should be a easy fix,but I'm not real sure. I will try to look into it some when I have a little time.

**Also**
![device-2018-09-30-210844](https://user-images.githubusercontent.com/15154102/46261599-26fe5080-c4f6-11e8-858e-a80ca46ea578.png)

Anyone with any dev abilities want to help me try to get this going? At this point not much works. I have already fixed a few things and spent a fair bit of my free time on it in the last week. It only builds with a bunch of repo picks and a few hacks at this point. I just don't have tons of time.I may be able to do it alone (or not) but it will take forever if so.We could create a telegram group or something if any others would be willing to help out. I haven't heard from anyone in the other telegram group in a long time. Tobias (hennymcc) included,so I can only guess that after bringing many Android versions to our device he may be done. Not speaking for him of course,just a guess.
Either way,if anyone is interested in helping it would be great!
I will probably be without internet access for a few days to a week (holiday) So hopefully when I check back here I'll get some replies :)

**Comment by friedensfurz** [comment]
Nice to have you all back! 
@Mikahlo From the Lineage 15.1 Thread (https://forum.xda-developers.com/showpost.php?p=77308985&postcount=469). That might be really helpful as it seems to be just a permission issue from android and is (hopefully?) easily fixable 

@Spookcity That's really nice! I'd love to give you a hand with that, time is not my primary problem, but I don't have too much knowledge about that stuff.. However, the commits that were made to the devicetree when a new Android version came out were not too complicated. I just don't understand (yet :) ) why there must be any changes to the devicetree. Can't it just work with a new Android sourcecode and unchanged device repos?

Also, I have recently uploaded some patches to the kernel against CVE: https://github.com/friedensfurz/android_kernel_samsung_kminilte with the help of @cschsz

Happy developing ;)

**Comment by Mikahlo** [comment]
@friedensfurz : Thanks for the hint! I have no clue how to add apps manually to a group in android but it should be possible to "just" install some apps as system apps at least a file manager. Will try that out.

@Spookcity : There is no need to fix this because most users didn't even noticed this strange behavior. Regarding android 9 I'm obviously not able to help. Nowadays time is to short to even get started with compiling stuff :(

**Comment by friedensfurz** [comment]
@Mikahlo Moving the non-working apps in /system works, but takes up too much space for me, sadly. There is probably an easy fix via kernel commit though.

**Comment by Spookcity** [comment]
I pushed my Android 9 branches and opened a new issue on here for discussion of this. Please go [Here](https://github.com/Spookcity/ROMS-G800F/issues/26) If interested in helping out. Thanks!

**Comment by rodman01** [comment]
Great and many thanks

**Comment by aelfinn** [comment]
@Spookcity As I said on XDA, your XenonHD build has been rock solid for multiple people I migrated to it, plus it is fast and smooth. If you could perhaps keep that ROM loosely updated, that would be enormously appreciated! :)

**Comment by Spookcity** [comment]
@aelfinn Interesting about XenonHD. Thats a ROM I didn't use too much. Glad to hear though. I'm sure I can find some time over the next few weeks to build an update. AS long as there are no issues,it shouldn't be a problem.

**Comment by fionnb** [comment]
@Spookcity I'd like to support the request of @aelfinn . I installed your XenonHD build on 2 S5minis for my kids and apart from some really, really minor quirks it performs outstandingly stable and well.

**Comment by Spookcity** [comment]
@fionnb I'm glad you brought this up again. I actually forgot. I may have some time today or tomorrow to grab the sources and attempt a build. If successful I will post it here in the coming days.

**Comment by aelfinn** [comment]
@Spookcity: Thanks a lot in advance! :)

Give us a shout if you could use some bug-hunting help! I have a spare lying around, and I’d be happy to help.

**Comment by fionnb** [comment]
Actually I do have an issue on one of the XenonHD minis now: calendar provider and contacts provider are both regularly (about every 2 days on average) locking up and all apps that try to use them, too. Force closing them from the Apps Config immediately makes them work again. I already cleaned their caches and data, no no avail. All their data is synced in via carddav/caldav apps. I have installed gapps nano but disabled everything that would sync any personal data with google (except play store). This combo works for me on numerous other phones. I do not know, however, if those c*-provider apps are from LOS or from the gapps package. Do you know?

**Comment by Spookcity** [comment]
@fionnb Yes, Calendar Provider and Contacts Provider are ROM related apps I believe. I have built an update,so hopefully those issues will be solved. I ended up needing to build this in virtual box as I use Arch Linux and I suppose since it is a bleeding edge distro that some package(s) are broken for building Oreo ROMS. I couldn't get jack-server to start no matter what I tried. Either way,hopefully it turned out OK. It is completely untested.

**DOWNLOAD** - [XenonHD-190113-Unofficial-kminilte](https://www.androidfilehost.com/?fid=11410963190603901040)

Enjoy!

**Comment by Spookcity** [comment]
I built AEX 6.4 a bit ago and kept meaning to post it on here,but forgot. So here it is if anyone wishes to give it a go.


Edit:
Forgot this is a microG build,so it has everything needed for that.
[AospExtended-v6.4](https://www.androidfilehost.com/?fid=1395089523397925308)

**Comment by Spookcity** [comment]
I overclocked our kernel from 1.4 - 1.6. I also enabled the lower frequencies (200 and 300mhz) Figured some might wish to try it out.

**Download** [1.6 overclock](https://drive.google.com/open?id=1POI-tW_NLXOdZEkzSp1nySgygzYErg-_)

**Comment by Oleggluk** [comment]
@Spookcity Thank you so much for everything! I broke the Samsung in smithereens (hooray hooray hooray, I'm so glad))). I ordered Xiaomi Redmi Note 7 (4/64) for aliexpress. Good luck to everyone, thank you all, I say goodbye to you.

**Comment by Spookcity** [comment]
@Oleggluk You are most welcome. I hope I was able to make your experience using our little phone a bit more enjoyable...Or at least more interesting.Haha. Thank you for always being a willing tester!
I have been due for a new phone for ages now,but am hesitant because I've just spent so much time messing with this phone. I will probably use it until it breaks in little pieces like yours.
Enjoy your Xiaomi Redmi Note 7 and good luck to you in the future too!

**Comment by aelfinn** [comment]
@Spookcity
The Pie kernel seems to work well. Haven’t been able to test extensively, especially not wrt battery life.
Your AEX-6.4 build isn’t quite up to scratch, I think. Hardware buttons are not working (and the setting to toggle them on/off is missing from settings, even though searching Settings gives it as a result), Tapping on Ambient Display in Settings crashes the Settings app, and the Quick Settings edit button isn’t working.

**Comment by MikeTeh** [comment]
Hardware buttons are working for me, but I'm missing the hardware button configuration. I'd configured me navbar with most used functions.
I think Digital wellbeing needs a configured microg (use ROM only with f-droid)?

**Comment by Spookcity** [comment]
@aelfinn I also had no issues with the HW buttons on AEX-6.4. I also had no issues with the quick settings edit button,so I'm not real sure what the issue is for you. The ambient display crash happens on LOS 16 as well. I am still not sure why,but am looking into it.
@MikeTeh Yeah, I also noticed that Digital wellbeing doesn't work. Not sure why. I will rebuild in the near future and see if it is solved.

**Comment by fionnb** [comment]
@Spookcity Thanks a lot for the January XenonHD Update!
It works pretty well so far (4 weeks, no problems).

**Comment by Spookcity** [comment]
I built an update for AEX. Latest sources (device and AEX)
**Download** - [AospExtended-6.4](https://www.androidfilehost.com/?fid=1395089523397942948)

I also built AICP. This used to be based on Lineage,but is now based on GZOSP. **Not tested**
**Download** - [AICP-14.0](https://www.androidfilehost.com/?fid=1395089523397942965)

Enjoy!

**Comment by brAvEiitoOr** [comment]
Thank you Spookcity for the ROMs!

I wonder if you can build a no-microG version. I could try to build myself, but I have an old PC

**Comment by Spookcity** [comment]
@brAvEiitoOr Yeah, I can do that no problem. Sorry,I only use microG, so it's just something I just haven't thought to do. Next time I build AEX ( probably a few weeks) I will build a non microG version. AICP will be longer as I don't have the sources anymore, bit I will do that as well

**Comment by Spookcity** [comment]
**AEX update**

- Synced AEX sources
- All recent device changes (including working HWC)
- Standard build and microG build

**Download**
[AEX-6.5 Standard build](https://www.androidfilehost.com/?fid=1395089523397965131)

[AEX-6.5 microG build](https://www.androidfilehost.com/?fid=1395089523397965132)

Enjoy!

**Comment by MikeTeh** [comment]
Thanks for the new AEX-builds.
Short reply, I'd installed microG build dirty over version from march,  works fine so far, but two things: 
-sms-app was gone (restored with TitaniumBackup)
-I've soft reboots when device is running under high load, fast actions in browser or gallery ie.
If I've more time I'll do a clean install, on the weekend I hope..
AICP 14 works fine at the other device, only home button won't wake the device and volume keys control the media volume instead of the ring tone.

Thanks again for your hard work on our device!

**Comment by Spookcity** [comment]
@MikeTeh Thanks for the feedback. Yeah,I had a random reboot on LOS 16. I think this is still related to HWC. Perhaps it is worse on AEX? I'm not sure. If it is you can just disable hardware overlays in dev options for now. Strange about the messaging app missing. Perhaps I did something wrong during the build. I'll look into it. I'll also try to rebuild AICP at some point and try to fix the home button wake.

Also,I ran across this ROM a few weeks ago and decided to build. It has pretty much every customization option I think one could want. Anyway,here it is. I've not tested extensively.

**Bootleggers ROM - 4.4.1**

[Standard build](https://www.androidfilehost.com/?fid=1395089523397969245)

[MicroG build](https://www.androidfilehost.com/?fid=1395089523397969246)

Enjoy!

**Comment by MikeTeh** [comment]
@Spookcity I hadn't the time for a clean install AEX  at my main device (disable hardware overlay in dev options changed nothing, freeze and reboot are still there), but I tried Bootleggers ROM with gapps at the second device. Works really smooth, no reboots, it's a nice ROM, I'd never this ROM before. Only I lost my IMEI, hope stock ROM bring it back... ..tomorrow. 
Then I'll flash also AEX clean.

**Comment by Spookcity** [comment]
@MikeTeh I agree,AEX seems to have some issues. I am unsure why. There were no problems while building,but along with reboots (I've had a few while using it too) It also just feels somewhat sluggish to me. I Don't know. I will rebuild sometime in the future and see if these issues are corrected.
I also agree about bootleggers ROM. I had never heard of this before either,but have been using it the last few days and I really like. It seems very stable as well. Weird about your IMEI. I hope you can sort that out!

**Comment by aelfinn** [comment]
@MikeTeh How did you lose your IMEI? And what is it now? I have a device too whose IMEI changed to something that appears to get blocked by phone service providers. Haven’t found a way yet to restore it to the original IMEI… :(

**Comment by MikeTeh** [comment]
@Spookcity First: Flash from Stock ROM brought me back my correct IMEI.
I will try it again on the second device with Bootlegger, maybe with MicroG build.
Thanks for your information about AEX 6.5, a clean install probably won't do much.
The next two days I am on business, in that time I think about going back to AEX 6.4 or maybe try crdroid for now.  ..or try another:)
In any case thank you for all your efforts!

Finally two curious side notes: 
-In all AEX Pie ROMs, a backup with Titanium Backup resets the wallpaper here.
-file upload in xenforo only works with Firefox and Opera

@aelfinn I made a clean installation of Bootleggers ROM with Magisk 19.2 and gapps pico.
How did you change your IMEI?

**Comment by MikeTeh** [comment]
Clean install only standard build Bootleggers ROM -> unknown IMEI again :(

**Comment by aelfinn** [comment]
@MikeTeh – I have no idea how the IMEI of that phone changed. It ran LOS 14.1 for the longest time, then I kind of decommissioned the phone and used it as a testing device. Maybe one or another of the ROMs I tested triggered that change. In my case, it’s a well-known IMEI, though: 350695200123951. I have tried flashing a stock ROM, but that didn’t help. Which stock version did you use, exactly?

**Comment by Spookcity** [comment]
Man,that's so weird about your IMEI. @MikeTeh I highly suggest if you haven't already that you backup your EFS partition in TWRP. I just wiped and flashed bootleggers ROM again out of curiosity and still no issues with my IMEI. I have never had this happen with all of the ROM flashing that I've done. I Just don't get how we are all using the same device,yet this happens.
@aelfinn I remember talking to you awhile ago about your IMEI. I'm sorry to hear that you haven't been able to get that fixed. I wish I could offer you more help,but I really have no idea.

**Comment by MikeTeh** [comment]
@Spookcity I swim in backups :)
Restoring only the efs (with TWRP 3.3.0-0) yesterday didn't bring any IMEI back into Bootlegger ROM, I installed Stock ROM, IMEI came back and did a new Nandbackup. 
I restored this complete backup after the second attempt, IMEI came back again.
I tried it with MicroG build - same result, unknown IMEI, very curious, because it works for you.
The problem of the second device- I wrote it here or in xda - it loses the connection to the mobile network after about 10-20 sec after the boot, with luck a call succeeds, so I can't test any SIM functions with it (the IMEI of the device matches the IMEI on the package, I also have a s4mini - one of three - with the same problem).
Maybe I'll try it again the next days on my main device, wish me luck :))

@aelfinn My main device had CPL1, later baseband updated to CRG3, the second device was already on CRG3 -CPL1 before - because I had the hope to get rid of the mobile network error by stock ROM flash)
I always check IMEI first after the ROM flash, I fear the change to another IMEI as it happened with you.
Anyway, I know the IMEI recovery is a bit complicated, but you should only flash the stock ROM in front of it. The IMEI can be changed, but the certificate is crucial.  This is readable, but you can lose it by wrong procedure. This is what I know about this problem.
Addendum: There are some devices on the net, including SM-G800F with the IMEI you specified. Did you see that?



Translated with www.DeepL.com/Translator

**Comment by aelfinn** [comment]
@MikeTeh – Yes, I have researched that problem for a while. My device seems to have changed to a blacklisted IMEI. I tried flashing the CRG3 stock ROM (via the latest heimdall on Linux), but that didn’t bring the phone’s original IMEI back. You say that “IMEI recovery is a bit complicated”; do you have any suggestions for what I might try in order to restore the original IMEI?

**Comment by MikeTeh** [comment]
@aelfinn I think there is nothing you can do yourself, ask here, I know him as really competent in these things, more I can not do for you
https://www.mhs-solutions.com/

@all sorry for OT

**Comment by MikeTeh** [comment]
@Spookcity Update TWRP to 3.3.1-0 and lean install cr droid https://forum.xda-developers.com/galaxy-s5-mini/development/rom-crdroid-5-0-pie-t3880298 and IMEI is allready there.
Bootlegger is the first ROM with this issue for me, crazy.

Happy weekend at all!

**Comment by Spookcity** [comment]
@MikeTeh Yeah that is so strange. I really have no idea what it could be or why it works without issue for me and not you. Also not sure why just this particular ROM. Just one of those bizarre things I guess.
Happy weekend to you as well!

**Comment by rodman01** [comment]
hi @Spookcity do you know whether and how I can force or verify the reboots? I have added graphics.composer to smdk3470-common.mk and do not have reboots so far. Also the ui feels a bit smoother and faster. I tried videos from youtube and google play film, they are working and no reboots so far. So I am not sure whether the include changed something. Also I am wondering whether we maybe need also additional the hwcomposer.xxx? How would be the right file name?

**Comment by Spookcity** [comment]
Hey @rodman01 First,thanks a lot for taking the time to work on this. I really appreciate it. I thought about perhaps needing to add a hwc.cpp (or multiple) cpp files to our repo as I've seen this in other devices. I didn't get any further than thinking about it though haha.
As far as forcing or verifying reboots I'm really not sure. It seems to be rather sporadic. I've had about 5 over the last week. All seem to be when I'm using the browser,but nothing more specific really and I haven't been able to grab a log. If you don't apply [This patch](https://github.com/Spookcity/android_device_samsung_smdk3470-common/blob/P/patch/hardware_interfaces/0002-libhwc2on1adapter-fix-up-layer-handles.patch) it will be very easy though. you will get reboots anytime you unlock your sim,try to sign into WiFi,or if you go to settings _> about phone and then android version and click it a few times to bring up the P easter egg you should trigger a reboot with this as well.
I just did a build with [These](https://review.lineageos.org/q/topic:%22no-skipvalidate%22+(status:open%20OR%20status:merged)) patches applies by hand. I have tried before and they didnt fix it,but I was hoping maybe along with the libhwc2on1adapter patch maybe that would do the trick. I have not tested yet though.

**Comment by rodman01** [comment]
Many thanks for the info and yes such reboots (most of them were soft reboots as far as I remember) I also noticed some times....Yesterday I mixed up something during my changes and what broke my wlan. So I couldn't get a connection anymore gggr :-( so I reverted everything step by step and sure how it is in most of such cases, only the very last revert brought my wifi back. Anyways I ll add graphics composer service and a few other things and give it a new try :-) I will also add the commits from leos you linked. Will see whether we could narrow down the problem a bit.
To all: I will upload my RR Testversion and if it makes sense a few test roms too, ist someone here who could test it (them) to verify whether something has changed and maybe fixed? Or should I better post it on xda in the RR thread?

**Comment by rodman01** [comment]
OK I have made an RR Test build with some changes and its uploaded. I haven't tested it much and not grapped a logcat yet, so if somebody has the some time and motivation, I would be glad if we can get some info. As for now, I have no major issues.
What I did and changed (will update my github sources, once I have sorted out the commits and when there are no bugs and changes needs to be reverted):
- synched hw samsung with latest leos commits
- in hidl manifest: included impl level to generic at graphics allocator
- in hidl manifest: changed graphic composer to hwbinder
- in hidl manifest: included impl level to generic at hardware wifi
- in common.mk: included graphics composer but not in hidl manifest (I think then it makes no sense at all :-)
- in overlay config.xml included some gps and wifi softapp, as well as hwc overlays, also dozealwaysonenabled it set to false
- in overlay powerprofile included some settings found on other device, cannot remember which one sorry
- included this 2 commits spook [mentioned](https://review.lineageos.org/q/topic:%22no-skipvalidate%22+(status:open%20OR%20status:merged))
- included the wifi patch form [here](https://github.com/Exynos5433/android_device_samsung_gts2-common/blob/lineage-16.0/patches/hardware_broadcom_wlan.patch)

Hope I haven't forgotten to mention something :-)

So please test WIFI stability, for example disconnecting after going in deep sleep), GPS (I am getting gps fix after less than a minute with gps status, but only after reboot after a clean installation) and of course possible reboots caused by HWC

I did a complete new flash and installed also opengapps nano, please note that mobile data is disabled by default in system.prop. Rom is based, beside the changes above, on last weeks RR and LEOS sources.

And here is the link, so download and have fun :-)
TEST_RR-P-v7.0.2-20190526-kminilte-Unofficial.zip: https://www.androidfilehost.com/?fid=6006931924117881841

**Comment by Spookcity** [comment]
@rodman01 I will gladly test out your RR build. I cant really  test WiFi as I have never had any issues with it,but I will test other things.
 As far as the SkipValidate patches. Sadly they do not change anything. I had a soft reboot when I was out and about watching YouTube videos via the browser app. I then got home and connected to adb to try to reproduce this and of course couldn't. It just wouldn't reboot. Its a very random issue it seems.
Thanks again for your work. I will let you know my findings.
 
Please do share your updated github sources whenver you have the time cause I will add some of your changes to my main repos if all seems OK. Thanks again. It is really appreciated!

**Comment by rodman01** [comment]
Yes seems that you are right. Youtube browser app ist starting and videos also begin to play As soon as I tab on full screen mode, the app freezes and doesn't go back. App stops and there is a kind of soft reboot. I will revert the 2 commits and make a new build and upload this is a few. Google play movie instead is changing to full screen and film is running but reverting back to the film list is not smooth, its more or similar to freezing.

**Comment by Spookcity** [comment]
Yeah,I'm not sure what the cause is. I will try to grab a log later.... First things Ive noticed about your RR build in just a minute of using...
LiveDisplay works! Did you do something specific for this? Is it perhaps just the updated hardware/samsung sources. I couldn't get this to work for anything.
Also noticed that the camera works (as well as before) I thought I remember the main reason i switched our graphic composer to passthrough was because hwbinder broke the camera. Either way,it seems to work now! I will test more and report back later.

**Comment by rodman01** [comment]
No I added nothing about livedisplay additional to the latest leos commits.
What I also added to boardconfigcommon.mk are this lines which I took from [here](https://github.com/Exynos5433), but I guess that they are useless anyways because not relevant for our device.
`USE_OPENGL_RENDERER := true
NUM_FRAMEBUFFER_SURFACE_BUFFERS := 3

#Gralloc
TARGET_USES_GRALLOC1_ADAPTER := true

#(G)SCALER
BOARD_USES_SCALER := true
BOARD_USES_DT := true

#HWCServices
BOARD_USES_HWC_SERVICES := true
SKIP_DISPLAY_BLANK_CTRL := true`

**Comment by rodman01** [comment]
also with the reverted commits SkipValidate I have the same problems with youtube, its almost freezing when changing to fullscreen and so to horizontal view and back to vertical. But I only did a dirty flash with the updated rom version, will try a clean install later.

**Comment by rodman01** [comment]
New Test rom is uploaded: https://www.androidfilehost.com/?fid=6006931924117881894

**Comment by Spookcity** [comment]
@rodman01 Thanks for the new build. Will flash and test some. Good to know about LiveDisplay. Hopefully that means it will work on LOS as well with updated sources.

USE_OPENGL_RENDERER := true
NUM_FRAMEBUFFER_SURFACE_BUFFERS := 3
We had these two flags originally,but I removed them as I thought they were not needed and/or deprecated. I havent noticed any difference without them. I am not real sure about the others honestly. I think I remember trying the "TARGET_USES_GRALLOC1_ADAPTER := true" one,but didn't notice any changes. I'm not positive though.

I've not installed GApps on this ROM,so i am not sure about YouTube videos. Videos work in the browser though. Can't say if it will cause soft reboots though. I will continue to test.

**Comment by rodman01** [comment]
Yes can confirm youtube in browser is working much better as the youtube app.

**Comment by rodman01** [comment]
my repos are updated, but I reverted all the entries made in boradconfigcommon.
https://github.com/rodman01/android_device_samsung_smdk3470-common
https://github.com/rodman01/android_device_samsung_kminilte

**Comment by Spookcity** [comment]
@rodman01 Great,thank you very much! I will build soon with some (all) of your commits and see how things work out. Then if all is good I will cherry-pick them. Especially the power profile update. Thanks for that. That is something that I have been meaning to do,but kept forgetting as I knew that this changed in Pie.
Your RR build is running good for me. I will keep testing. Strange that videos with the YouTube app are kinda messed up. Not sure what is causing this.

**Comment by MikeTeh** [comment]
@rodman01 @Spookcity
Testbuid2 seems to be a good build, smooth, no reboots so far.
WLAN Hotspot Error, disables itself, error message under settings - hotspot and tethering :(
I don't use localizations, so what's the shortest way to test gps?
Localization and mobile data are disabled with installation.
youtube vanced brakes the device to standstill, "system ui does not react", only (soft)reboot helps (works in crdroid 5.4 ie)

Hotspot:
05-27 03:05:07.687 V/WifiManager(2673): registerSoftApCallback: callback=com.android.systemui.statusbar.policy.HotspotControllerImpl@ca7c317, handler=Handler (android.os.Handler) {ba94304}
05-27 03:05:07.713 V/WifiManager(2673): SoftApCallbackProxy: onStateChanged: state=14, failureReason=0
05-27 03:05:07.714 V/WifiManager(2673): SoftApCallbackProxy: onNumClientsChanged: numClients=0
05-27 03:05:10.294 I/WifiService(2271): startSoftAp uid=1000
05-27 03:05:10.296 D/WifiStateMachinePrime(2271): Starting SoftApModeManager
05-27 03:05:10.303 I/android.hardware.wifi@1.0-service(2072): Wifi HAL started
05-27 03:05:10.308 I/WifiP2pNative(2271): Registering for interface available listener
05-27 03:05:10.307 I/WifiVendorHal(2271): Vendor Hal started successfully
05-27 03:05:10.323 D/WifiP2pNative(2271): P2P InterfaceAvailableListener true
05-27 03:05:10.323 D/WifiP2pService(2271): Wifi enabled=false, P2P Interface availability=true, Number of clients=0
05-27 03:05:10.325 D/WifiP2pService(2271): Wifi enabled=false, P2P Interface availability=true
05-27 03:05:10.340 I/e       (2072): type=1400 audit(0.0:1671): avc: denied { write } for name="firmware_path" dev="sysfs" ino=1738 scontext=u:r:hal_wifi_default:s0 tcontext=u:object_r:sysfs:s0 tclass=file permissive=1
05-27 03:05:10.340 W/e       (2072): type=1300 audit(0.0:1671): arch=40000028 syscall=322 per=800008 success=yes exit=6 a0=ffffff9c a1=b068f5cc a2=20001 a3=0 items=1 ppid=1 auid=4294967295 uid=1010 gid=1010 euid=1010 suid=1010 fsuid=1010 egid=1010 sgid=1010 fsgid=1010 tty=(none) ses=4294967295 exe="/system/vendor/bin/hw/android.hardware.wifi@1.0-service" subj=u:r:hal_wifi_default:s0 key=(null)
05-27 03:05:10.340 W/auditd  (2038): type=1327 audit(0.0:1671): proctitle="/vendor/bin/hw/android.hardware.wifi@1.0-service"
05-27 03:05:10.345 I/e       (2072): type=1400 audit(0.0:1672): avc: denied { search } for name="wifi" dev="mmcblk0p3" ino=21 scontext=u:r:hal_wifi_default:s0 tcontext=u:object_r:wifi_efs_file:s0 tclass=dir permissive=1
05-27 03:05:10.345 I/e       (2072): type=1400 audit(0.0:1672): avc: denied { read } for name=".mac.info" dev="mmcblk0p3" ino=22 scontext=u:r:hal_wifi_default:s0 tcontext=u:object_r:wifi_data_file:s0 tclass=file permissive=1
05-27 03:05:10.345 I/e       (2072): type=1400 audit(0.0:1672): avc: denied { open } for name=".mac.info" dev="mmcblk0p3" ino=22 scontext=u:r:hal_wifi_default:s0 tcontext=u:object_r:wifi_data_file:s0 tclass=file permissive=1
05-27 03:05:11.532 I/WifiHAL (2072): Initializing wifi
05-27 03:05:11.532 I/WifiHAL (2072): Creating socket
05-27 03:05:10.345 W/e       (2072): type=1300 audit(0.0:1672): arch=40000028 syscall=54 per=800008 success=yes exit=0 a0=6 a1=8914 a2=be91d204 a3=be91d1ec items=5 ppid=1 auid=4294967295 uid=1010 gid=1010 euid=1010 suid=1010 fsuid=1010 egid=1010 sgid=1010 fsgid=1010 tty=(none) ses=4294967295 exe="/system/vendor/bin/hw/android.hardware.wifi@1.0-service" subj=u:r:hal_wifi_default:s0 key=(null)
05-27 03:05:11.547 I/WifiHAL (2072): Initialized Wifi HAL Successfully; vendor cmd = 103
05-27 03:05:11.548 I/android.hardware.wifi@1.0-service(2072): Adding interface handle for wlan0
05-27 03:05:11.548 I/android.hardware.wifi@1.0-service(2072): Adding interface handle for p2p0
05-27 03:05:11.548 E/android.hardware.wifi@1.0-service(2072): Failed to register radio mode change callback
05-27 03:05:11.548 I/android.hardware.wifi@1.0-service(2072): Configured chip in mode 1
05-27 03:05:11.570 D/WifiP2pNative(2271): P2P InterfaceAvailableListener false
05-27 03:05:11.570 D/WifiP2pService(2271): Wifi enabled=false, P2P Interface availability=false
05-27 03:05:11.582 D/WificondControl(2271): Setting up interface for soft ap mode
05-27 03:05:11.592 I/WifiNative(2271): Interface state changed on Iface:{Name=wlan0,Id=17,Type=AP}, isUp=true
05-27 03:05:11.592 I/WifiNative(2271): Successfully setup Iface:{Name=wlan0,Id=17,Type=AP}
05-27 03:05:11.593 V/WifiManager(2673): SoftApCallbackProxy: onStateChanged: state=12, failureReason=0
05-27 03:05:11.662 W/BroadcastQueue(2271): Background execution not allowed: receiving Intent { act=android.net.wifi.WIFI_AP_STATE_CHANGED flg=0x4000010 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
05-27 03:05:11.667 D/WifiService(2271): handleWifiApStateChange: currentState=12 previousState=11 errorCode= -1 ifaceName=wlan0 mode=1
05-27 03:05:11.676 E/WifiVendorHal(2271): setCountryCodeHal(l.1831) failed {.code = ERROR_NOT_AVAILABLE, .description = }
05-27 03:05:11.681 W/wificond(2128): Copy constructor is only used for unit tests
05-27 03:05:11.682 W/wificond(2128): Failed to get NL80211_ATTR_EXT_FEATURES
05-27 03:05:11.820 I/android_os_HwBinder(2271): HwBinder: Starting thread pool for default::android.hardware.wifi.hostapd@1.0::IHostapd
05-27 03:05:11.991 E/WifiNative(2271): Failed to add acccess point
05-27 03:05:11.993 V/WifiManager(2673): SoftApCallbackProxy: onStateChanged: state=14, failureReason=0
05-27 03:05:11.994 E/WifiController(2271): SoftAP start failed
05-27 03:05:11.995 D/WifiService(2271): handleWifiApStateChange: currentState=14 previousState=12 errorCode= 0 ifaceName=wlan0 mode=1
05-27 03:05:11.995 D/WifiService(2271): updateInterfaceIpState: ifaceName=null mode=-1 previous mode= -1
05-27 03:05:11.995 I/WifiService(2271): stopSoftAp uid=1000
05-27 03:05:11.996 W/BroadcastQueue(2271): Background execution not allowed: receiving Intent { act=android.net.wifi.WIFI_AP_STATE_CHANGED flg=0x4000010 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
05-27 03:05:12.210 D/WificondControl(2271): tearing down interfaces in wificond
05-27 03:05:12.216 E/WifiHAL (2072): wifi_event_loop: Read after POLL returned 4, error no = 0 (Success)
05-27 03:05:12.216 D/WifiHAL (2072): Got a signal to exit!!!
05-27 03:05:12.216 I/WifiHAL (2072): Exit wifi_event_loop
05-27 03:05:12.216 E/WifiHAL (2072): wifi_cleanup: Read after POLL returned 4, error no = 0 (Success)
05-27 03:05:12.216 E/WifiHAL (2072): Event processing terminated
05-27 03:05:12.216 I/android.hardware.wifi@1.0-service(2072): Legacy HAL stop complete callback received
05-27 03:05:12.217 I/WifiHAL (2072): Internal cleanup completed
05-27 03:05:12.419 I/android.hardware.wifi@1.0-service(2072): Wifi HAL stopped
05-27 03:05:12.420 I/WifiVendorHal(2271): Vendor Hal stopped
05-27 03:05:12.421 I/WifiNative(2271): Successfully torn down Iface:{Name=wlan0,Id=17,Type=AP}
05-27 03:05:12.423 I/WifiNative(2271): Successfully initiated teardown for iface=wlan0
05-27 03:05:12.424 V/BatteryExternalStatsWorker(2271): no controller energy info supplied for wifi
05-27 03:05:13.171 V/WifiManager(2673): registerSoftApCallback: callback=com.android.systemui.statusbar.policy.HotspotControllerImpl@ca7c317, handler=Handler (android.os.Handler) {ba94304}
05-27 03:05:13.173 V/WifiManager(2673): SoftApCallbackProxy: onStateChanged: state=14, failureReason=0
05-27 03:05:13.174 V/WifiManager(2673): SoftApCallbackProxy: onNumClientsChanged: numClients=0

**Comment by rodman01** [comment]
sorry I wasn't online yesterday anymore...
@Spookcity yes the youtube issue is strange, because after I reverted everything, I haven't had this problem.
@MikeTeh great many thanks for the log. Yes I also noticed some problems with wifi disconnections. But I am not using the hotspot. So the problem might be related to the changes in overlay:
`
    <bool translatable="false" name="config_wifi_softap_acs_supported">true</bool>
    
    <bool translatable="false" name="config_wifi_softap_ieee80211ac_supported">true</bool>`
I will reverse them in next build and see whether it is fixed.

**Comment by Spookcity** [comment]
Yeah @rodman01 strange issue with YouTube indeed. Perhaps it was related to switching graphic composer to hwbinder. I can't really think of what else it would be. I've had 2 reboots since using your RR build sadly. So it seems as though this issue is still there. I wasn't able to grab a log. Hopefully at some point I will be able to. It seems,at least for me,that HWC works about 95% of the time without problems. I am just not sure what the remaining issue might be.

A bit off topic,but LiveDisplay just refuses to work in LOS. I have no idea why. It also now works in AEX. I assume they have just added this feature to their frameworks/base..Which is prob what RR did as well,so I guess thats why it works there. I just get a black screen on boot anytime I build any version of LiveDisplay for Lineage. kinda confusing and frustrating.

**Comment by rodman01** [comment]
ggr I thought too, that we are on a good way. I haven't had any reboots so far anymore, but do not use the mobile as daily driver, only when I flash new things and for testing. Anyways. You are right, after reverting the composer back to passthrough the youtube app returns to normal :-) Switching from portrait to landscape is working again and is quite fast again. After I changed other things by try and error previously, I got the change too, but very slow, but the app was not crashing anymore. Not even freezing, only slow movements. Also systemui didn't stopped anymore.
What I also did is set treble omx in system.prop to false which I found for example [here](https://review.lineageos.org/c/LineageOS/android_device_nvidia_foster/+/246450/8/system.prop#b18).
What I also did is  [this](https://github.com/ivanmeler/android_kernel_samsung_herolte/commit/3b86944b93f291cb116473b66640af54d0130394) kernel commit, but not sure whether this helped anything, but it works :-): Kbuild: Add missing videodev2 exynos headers.
And at least I copied the open max video board configurations from gts2-common(https://github.com/Exynos5433/android_device_samsung_gts2-common/blob/lineage-16.0/BoardConfigCommon.mk). Think that this alone is also not sufficient or maybe even wrong.
I will upload a new RR Testbuild in a few, if somebody is interested in a test

About livedisplay: usually RR doesn't add not RR specific comits to their synced or forked leo repos. Or better thats they used to do it in the past :-). And I haven't found any related commits at their fwb or somewhere, which is not also included in leos repos. Haggerty just merged liveisplay related commits in the [msm8976 common device repo](https://github.com/LineageOS/android_device_samsung_msm8976-common/commits/lineage-16.0). Maybe this helps and points to the right direction? Although I haven't included something of that in my device sources, as far as I remember :-)

**Comment by rodman01** [comment]
ok new RR test build is online. I also reverted the overlays relatet to wifi and softap. So if you can test also the hotspot again, it would be great.
Test3-RR-P-v7.0.2-20190529-kminilte-Unofficial.zip: https://www.androidfilehost.com/?fid=6006931924117884250

**Comment by MikeTeh** [comment]
Thanks, I'll try RR3, I had today the first two reboots in RR2 (no logs, connected with car over Bluetooth this time).
No other problems so far..

**Comment by Spookcity** [comment]
@rodman01 Thanks a lot for the info and again also for working on this. Yeah,the HWC issue is weird as it seems to work without issues most of the time,but then a few times causes soft reboots,so I'm not sure. I believe that the treble omx prop is deprecated as we had this at one point and I removed it. Not sure about the open max video ones,but I can't imagine they are going to hurt. Same with the videodev2 exynos headers kernel commit. Not real sure about this,but hey if it builds it probably doeasnt hurt anything.
I tried to update our CameraWrapper.cpp with [This commit](https://github.com/davidmueller13/device_lt03lte_nougat/commit/fe21be3e1b2705cebd54012c6f29c31d80d79570) but sadly it didnt change anything,so I'm really at a loss at to what exactly is wrong with our camera.

Thanks for the info on LiveDisplay. Haggerty's commit is pretty similar to one of the things that I have tried (except using the exynos version) I also tried building the LiveDisplay sysfs. It's weird because if I don't add LiveDisplay to out boardconfig and the HAL to our manifest it doesn't work and when I do I get a black screen. Thats why i assumed that RR added this to their fwb. Either way,its a minor thing. I would just like to get it working. I will contimue to mess with it when I have time.

I'll flash your new RR build later and report back my findings. Thanks!

**Comment by rodman01** [comment]
Maybe later today or tomorrow I can try to build a leos version with your sources too, just to verify and check how it is working on my device. Perhaps so we could get a hint about the not working livedisplay.

In the meantime I tried several other thing and at the moment with some commits included in openmax/osal RR is running good and youtube is almost smooth and fast :-) I took this: https://github.com/LineageOS/android_hardware_samsung_slsi-cm_openmax/commit/faf9a76b1e26a78d65d5b61ff5137aae63ad8a20
and some of this: https://github.com/Exynos5433/android_hardware_samsung_slsi-cm_exynos5433/commits/lineage-16.0. Will add the repo to my github and push the commits I cherry picked.

Currently and still no reboots :-). But I have 2 Warnings in logcat for which I cannot find something with google
W MapperHal: buffer descriptor with invalid usage bits 0x2000 (in youtube "section"
W MapperHal: buffer descriptor with invalid usage bits 0x400 (paired with)
E ion     : ioctl c0044901 failed with code -1: Invalid argument 

Camera I haven't tested yet.

**Comment by Spookcity** [comment]
@rodman01 Great,thanks! I think I remember seeing these errors in our logs for awhile now,so they may be (mostly) harmless hopefully. Interesting commits. Hopefully they will indeed help. Thanks a lot!

**Comment by rodman01** [comment]
github updated: https://github.com/rodman01/android_hardware_samsung_slsi_exynos3470
mmh I notice no significant change :-) so this is a good new, but also a bad :-)

But I haven't done a clean flash for a while, maybe its time for it now :-)

Edit: camera is crashing gggr, will do a reset and new installation now.

**Comment by rodman01** [comment]
Gave up the hwc thingis and camera problems for now :-), I am lost at the moment...
But synced the leos sources already as said and its building now. I noticed when I tried to merge the latest hardware/samsung leo commits to your repo, many conflicts. I set everything to theirs, and now its still building, but maybe that there is the reason why livedisplay is not working for you? Have you also merged the leo commits from 24. and 25.5?

**Comment by Spookcity** [comment]
Yeah,the HWC issue is confusing. I think I will just go ahead and disable HWC again in my next LOS build and revert the related commits. I honestly haven't really noticed any downside to this. Battery seems to be OK and the system seems pretty fluid with it disabled. 

As far as my hardware/samsung repo:
Yeah, I have merged the latest LOS hardware/samsung into my repo. I just keep forgetting to push. I did have merge conflicts I think because i had repo picked some LiveDisplay stuff that eventually became part of the repo,but in a different form. I solved these conflicts I think properly,but it is def possible that I messed something up along the way. I will wait to see what your results of are with LiveDisplay before I push because if it works for you then I def messed something up.
You are going to have some RIL problems since you are using LOS's hardware/samsung repo as there are patches applied to mine,but this will obviously not effect LiveDisplay. Thanks again for helping out!

**Comment by rodman01** [comment]
okidoki build finished fine and in my opinion live display is working. It is different to RR, if I remember right, there is one colour setting missing, but not sure. What is not working or greyed out is the colour settings the others are there but the effects are lower then on RR. But that is just a personal feeling.
Anyways here is the link, if you and all others what to test my leos build. I also used your hardware/samsung, but merged the leos 16 branche and solved the conflicts choosing: set the conflicted files to theirs (leos 16). If you want I can also put it on my github

lineage-16.0-20190530-UNOFFICIAL-kminilte.zip: https://www.androidfilehost.com/?fid=6006931924117884987

**Comment by Spookcity** [comment]
@rodman01 Ya know what? I decided to start a clean build when I got home as its been a bit since Ive done so....and LiveDisplay is indeed working for me now too! I have no idea why. I did add the commit you made to our overlay "config_setColorTransformAccelerated" I am not sure what that does honestly,but could this be it? Did you use this in your build as well or did you use my repo? Perhaps I will try to add livedisplay stuff to our manifest and BoardConfig and see what happens,but either way,its working now,so great! Thanks again for all of your work!
I will push my hardware/samsung repo and I cherry-picked about 20 kernel commits relating mostly to bluetooth and wireless. I will push this as well. Thanks again!!

Edit: All of my repos have been updated and pushed

**Comment by MikeTeh** [comment]
RR test3: Hotspot is working again, phone app is loosing favourites sometime.
I'd changed live display over the qs-tile from auto to night, off and on again and I had a reboot

05-31 18:23:03.648 I/LineageSystemServer(9712): Attempting to start service org.lineageos.platform.internal.display.LiveDisplayService
05-31 18:23:03.662 I/LineageSystemServer(9712): Starting service org.lineageos.platform.internal.display.LiveDisplayService
05-31 18:23:03.662 I/SystemServiceManager(9712): Starting org.lineageos.platform.internal.display.LiveDisplayService
05-31 18:23:21.544 W/hwservicemanager(1967): getTransport: Cannot find entry vendor.lineage.livedisplay@2.0::IDisplayModes/default in either framework or device manifest.
05-31 18:23:21.552 W/hwservicemanager(1967): getTransport: Cannot find entry vendor.lineage.livedisplay@2.0::IReadingEnhancement/default in either framework or device manifest.
05-31 18:23:21.572 W/hwservicemanager(1967): getTransport: Cannot find entry vendor.lineage.livedisplay@2.0::IColorBalance/default in either framework or device manifest.
05-31 18:23:21.578 W/hwservicemanager(1967): getTransport: Cannot find entry vendor.lineage.livedisplay@2.0::ISunlightEnhancement/default in either framework or device manifest.
05-31 18:23:21.591 W/hwservicemanager(1967): getTransport: Cannot find entry vendor.lineage.livedisplay@2.0::IPictureAdjustment/default in either framework or device manifest.
05-31 18:23:21.610 E/LiveDisplay(9712): Color temperature out of range: 0
05-31 18:23:21.620 W/SystemServiceManager(9712): Service org.lineageos.platform.internal.display.LiveDisplayService took 112 ms in onBootPhase

**Comment by rodman01** [comment]
hello and good morning!
@MikeTeh many thanks. I have added livedisplay-sdm and sysfs as well as advanced display to the device and have no reboot after doing the tests you mentioned above. I cannot get rid of the warnings in logcat, but I hope the reboots are gone for you too anymays.
New RR testbuild is up: [Test4-RR-P-v7.0.2-20190601-kminilte-Unofficial.zip](https://www.androidfilehost.com/?fid=6006931924117886626)
@Spookcity I have added the missing entries of livedisplay to hidl manifest, but depending on which one I all included, I get difficulties from black screen to not booting over endless phone is updating message. Do you know why? and/or what is to do to solve the warnings? Do we need to change to passthrough? I also notices some of this warnings for example related to audio.

My logcat seems to look not to bad now, but I have crash regarding to fingerprint validity service. I changed biometrics fingerprint to 2.1 but no change. Is fingerprint working in general with P, and with your leos P? Or is it currently broken in general. If you want to check my RR log, please find it attached.
Clean install of RR without wifi and without gapps.
[logcat.txt](https://github.com/Spookcity/ROMS-G800F/files/3244419/logcat.txt)

**Comment by Spookcity** [comment]
@rodman01 I have no idea about the LiveDisplay issue. This is the problem I was experiencing. I didn't grab any recent logs from the blackscreen caused by livedisplay,but older ones showed missing libs. No idea what that was about,but it was a bunch of them. For whatever reason it just doesn't seem to work if we include the entries in the manifest. Not sure why at this point.
I personally don't use the FP scanner,but I think that in general it works. The fatal errors are always there upon boot,but it seems to connect after a bit. I think (if i remember correctly) that the FP works decently at least,but there are issues with initially registering the fingerprint. This is something I really havent looked into at all. I do know that [This](https://github.com/AICP/device_samsung_klte-common/commit/2f41fd2d2ff6ba2a74572309d9e6858967b140d6) was something I wanted to look into,but just havent had the time recently.

**Comment by rodman01** [comment]
okidoki and thanks....i will have a look at it and try to cherry pick / "port" the commit.
Will see how it works :-)

**Comment by rodman01** [comment]
stupid question, because I cannot find the settings: where to enable fingerprint unlock?
I recently saw it in setup wizard and skipped it all the time, but in RR and without gapps I do not have a setup wizard. maybe I cannot find it because its not working and not properly included? But with the klte changes the fatal error in logcat disappeared, but others related to that appeared :-).
[logcat.txt](https://github.com/Spookcity/ROMS-G800F/files/3244929/logcat.txt)

edit: never mind I found it after flashing leos 16 spookies original and also in my leos build it is in settings available.

**Comment by Spookcity** [comment]
It should be under security and location if it built properly. Otherwise I'd guess something didn't work.
I'll have to have a look at your logcat when I'm at my laptop as in the fasthub app I can't for whatever reason.  

Edit:
You beat me to it :smile:

**Comment by aelfinn** [comment]
@Spookcity The overclocked kernel works well, btw. Would it be hard to enable the higher and lower frequencies in the default kernel of one of your ROMs, or does one have to recompile the kernel? Also, would it be possible to include a version of IntelliPlug?

**Comment by Spookcity** [comment]
@aelfinn Glad to hear its working out! I guess I could try to use anykernel2 so it would just work with any Pie ROM. I'm just not sure of any changes to anykernel and how easy/difficult it would be. I will look into it when I have some time. As far as adding IntelliPlug... It really should just be a matter of cherry-picking the commit since I already added it to my N_custom kernel. Again,I will look into this when I have some free time.

**Comment by MikeTeh** [comment]
I have tested RR Test4 for a few days now in everyday use and it works fine for me, no reboots, hotspot works again.
Is it planned to put it in xda?
Thanks for your work on it @rodman01

**Comment by rodman01** [comment]
Thanks for your feedback. I am currently in holiday, but after my return middle of next week, I will update my sources and try another and new build. Maybe I will put a release note in the current RR thread from @friedensfurz if he don‘t mind, but at the moment I am not planning to open a new thread of and for my own.

**Comment by Spookcity** [comment]
**AospExtended update**

This build seems to be running smoothly unlike  the previous one. I've been using it for a day plus. Also fixed the messaging app not being built (stupid mistake on my part last time)
Updated AEX sources. June security patch and same changes from the LineageOS 16 build.

**Download**
[Standard Build - AEX-6.6](https://www.androidfilehost.com/?fid=6006931924117896684)
[MicroG Build - AEX-6.6](https://www.androidfilehost.com/?fid=6006931924117896685)

Enjoy!

**Comment by aelfinn** [comment]
@Spookcity I’d be happy to test a kernel! :)

**Comment by Oleggluk** [comment]
@Spookcity Hi! Glad to see you are fine. I bought an RN7. I'm happy, 10 hours of screen work in 4G+. Thank you for everything that you did for Samsung, I will never come back))))). Join us). (Google translator)

**Comment by rodman01** [comment]
so finally is done :-) New RR build is up dan posted on xda. Here it is for your interest too.
Download
RR-P-v7.0.2-20190624-kminilte-Unofficial.zip: https://www.androidfilehost.com/?fid=6006931924117902206
https://forum.xda-developers.com/showpost.php?p=79788255&postcount=85
@Spookcity I have added 2 wifi commits to the kernel, which I found in leos review and are related to error lines in logcat of my previous build. Also in device repos I did a few changes wifi and media codecs related. All are or is uploaded to my github. Till now it seems to be stable and after my simple tests (no whats up, no instagram etc etc) only youtube (browser and app) and google play film rom seems to be stable.
What I still haven't got solved is the crash in logcat related to validity service, so I reverted all my tries and it is working with your code anyways :-). And the slow or very long lasting of gps fix in gps status. Maybe some other can test and verify whether gps is working, not working or working good.

edit: download link fixed

**Comment by Spookcity** [comment]
@Oleggluk Great to see you around here!! I hope all is well with you. Sounds like it definitely is when it comes to your phone. Ya know,I have been due a new device for well over a year now. It would actually save me a little money on my contract to get a new one,but I am so "out of the loop" on what phones are good and what-not at this point. I am ashamed to admit i needed to look up what the RN&7 was :) I will certainly keep that in mind whenever I finally do decide to move on from my mini.

@rodman01 Thanks for the RR build and also the info on your commits. I will look into adding some of these to my main repo as well. Especially the WiFi ones. I don't see how they would be anything but helpful,so thanks!

@aelfinn OK,I have built a P_custom kernel and used anykernel2. It should flash on any P ROM. I have basically just added things (optimization flags,CPU governors,IO schedulers etc) that were not a huge pain in the a$$ to bring over from my N_custom branch. When I have time I will try to get more added. I am not going to give a ton of detail about what the kernel includes,so just have a look [Here](https://github.com/Spookcity/android_kernel_samsung_kminilte/commits/P_custom) and you'll get a good idea. I used a linaro 7.2 toolchain to build this. I have tested for a few days and it seems stable fro me. Although Ive only really used the zzmoove governor and messed with voltage some. Anyway...Hope it works out and there is some noticed benefits.

**Download**
[SpookcityKernel-P](https://drive.google.com/open?id=1m0cg3PQV1XGGPSbBIi1QxuhTDaulX6EL)

**Comment by aelfinn** [comment]
@Spookcity Will try a.s.a.p.!

Incidentally, the one device in my circle of friends that is running your LOS16 build (2019-06-09) has sporadic reboots, which seem to be associated to incoming/outgoing calls in every instance. Any idea what might cause that or what to try?

**Comment by rodman01** [comment]
@Spookcity thanks for your new kernel and already downloaded :-) Working well also on RR build.
According to my kernel commits: I noticed a more stable connection with the changes, but I still have disconnections especially during the restore of apps and settings from google play. So from my point of view it is still not perfect.

**Comment by brAvEiitoOr** [comment]
Thanks @Spookcity for the kernel. Unfortunately, the phone can´t boot. Could you try to make an 1.5Ghz version, like in Nougat? Or how to solve it?. Thanks in advance

**Comment by Spookcity** [comment]
@aelfinn Thats a weird issue your friend is experiencing with the ROM. I honestly have no idea what the cause may be. I haven't had any reboots while the HWC has been disabled. Perhaps they can get a log to see exactly what is happening.

@rodman01 Yeah,it probably will be a series of commits that hopefully will eventually solve the WiFi issue,so these commits are def a step in the right direction. I haven't had any time lately to really look into anything,but hopefully i will have a bit more in the coming weeks.

@brAvEiitoOr So weird how the 1.6 overclock works for some but not all. I guess this is just something I will never know the reason for. I mean we are all using the same device with the same CPU. Very weird. Either way, i will do a 1.5 version whenever I have some time. It is not very difficult or time consuming,but right now I just really don't have any time. Hopefully within a week or 2 though.

**Comment by aelfinn** [comment]
@Spookcity She depends on the phone as her main device, so I can’t take it away from her for extended testing. Is there maybe an app that can help in such circumstances? (By constantly taking a log, of fixed length, until some trigger event such as a reboot?)

**Comment by Spookcity** [comment]
@aelfinn Ahh,yes I can understand this. I always used Matlog in the past. Although I haven't in some time as I usually just connect to adb shell now, but this isn't super user friendly. There is probably an option in Matlog for something that would be helpful though I would think. Root is required. 
Unless someone else has a better idea that's all I can really think of off hand.

**Comment by rodman01** [comment]
I think I have narrowed down my (the?) wifi disconnection problem. After lots of tries with fw path, wifi supplicant, kernel etc etc changes without success, I am quite sure that it is caused by my wlan router settings. My wlan has combined 2.4 and 5 GHz signal and it seems that as soon as one the connection or the speed is going down, device disconnects and reconnects to the than stronger signal. After setting up 2 different wlans (one 2.4 and a separate 5GHz) and connecting to the one or the other, the problems seems to be gone. I remembered that in some rom versions there had been the possibility of choosing the preferred network frequency in wifi advanced settings, but in pie it seems to be missing. I already included: dual_band_support in overlay, but I cannot find the setting either. Do I need to add some other commits to get the settings possibility available in wifi advanced settings? Every help is very appreciated :-)
`
    <bool name="config_wifi_dual_band_support">true</bool>`
![image](https://user-images.githubusercontent.com/5551595/61233945-1724c580-a732-11e9-9d79-c55d4333a2df.png)

**Comment by Spookcity** [comment]
@rodman01 First,great job tracking this down! This makes a lot of sense. It explains why some have this issue while others don't. It also explains why I don't have this issue as I'm using a very old router that is not very good and almost certainly only has the 2.4 frequency. Sadly I have no clue about how to get a selectable frequency within settings. It seems like this should def be possible. I will try to look into this when I have some time. Hopefully we can figure this out.

**Comment by rodman01** [comment]
I found a [commit](https://android.googlesource.com/platform/packages/apps/Settings/+/fa37a75%5E!/) to "Add frequency band control in UI" and included already the additionals in the xmls and it is available in wifi advanced settings and gives no error. But its currently doing nothing sure :-) what is difficult to do for me is to add the changes in the relevant java file. The wifi AdvancedSettings.java is not available anymore in pie. I would guess the changes needs to be inluded in configurewifisettings.java. I have tried it but it couldn't be done via copy and paste and my effort leads into building errors. Can anyone assist me? Thanks in advance.

**Comment by Spookcity** [comment]
@rodman01 Hey apologies for the rather late response,but I am away from home for the next week or so...I would think perhaps that this commit could be adjusted and added to the WifiSettings.java file or as you said the ConfigureWifiSettings.java within settings. However,my java skills are rather limited sadly so I'm not sure how much help I would actually be. I will gladly try whenever I am home and find some time though. It would be a bit of trial and error for me to figure it out.

**Comment by Spookcity** [comment]
@rodman01 I didn't forget about trying to get the frequency selection to be switchable in settings. I actually messed with it a bit about a week ago,but sadly had no luck. I was hoping to find some more time to mess with it,but so far I haven't been able to make any. Hopefully soon. Although I think I may lack the needed skill to get this ported over to Pie.

**Comment by rodman01** [comment]
@Spookcity don‘t worry it would be only an addon and a special thing and nice to have :-) I have also treid and messed around a bit but ended only in build errors. I guess the easiest would be to create an own advanced settings java file. But my knowledge is limited so I can only do it with advanced try and error :-) Currently i am in holiday maybe after next weekend when I am home again I start a new „action“ :-)

**Comment by Spookcity** [comment]
@rodman01 I agree. It would be a nice feature to have though. If I am able to find some time I will try to mess with it some more as well. Enjoy your holiday!

**Comment by fionnb** [comment]
Hi @Spookcity happy new year 2020. I'd like to, once again, kindly ask if you would mind to try another XenonHD build because the last one is now one year old. Otherwise I'd be happy to hear suggestions what I should try as an alternative.

**Comment by Spookcity** [comment]
Hey,Happy New Year to you as well @fionnb! I hope all is good.

Well,my time is super limited lately.Plus,I have recently bought a new device. This does not mean I'm done with the mini though. It's all about time really. I don't remember,but was it XenonHD Oreo or pie? I think I have issues building Oreo ROMS due to jack issues that I was unable to sort out last I tried,but I don't remember. I can try again and download the sources. 
I can say that almost certainly the security patches are going to be way behind on this ROM as it seems that most custom ROMS,aside from LOS have not updated their security patches in forever. I will give it a look when I find some time though.
